CMakeLists.txt

```diff
@@ -53,6 +53,14 @@ if (${CMAKE_SYSTEM} MATCHES "Darwin")
     set(EXTRA_LIBS ${CORESERVICES_LIB})
 endif()
 
+if (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang" OR ${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden")
+endif()
+
+if (${CMAKE_C_COMPILER_ID} STREQUAL "Clang" OR ${CMAKE_C_COMPILER_ID} STREQUAL "GNU")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden")
+endif()
+
```

Comment:
Setting `CMAKE_CXX_VISIBILITY_PRESET` to `hidden` should have the same effect.

Reply:
Did not know about those, thanks. I cannot see which version of cmake introduced them so I'm a bit cautious about using them.
That said, if @doxygen or others prefer that route I can respin.

Reply:
That was already availlable in CMake 3.0.0.

Reply:
[Here](https://github.com/doxygen/doxygen/blob/master/CMakeLists.txt#L14) it says that doxygen requires 2.8.12.

Reply:
That variable [was already in 2.8.12 as well][ref].

[ref]: https://cmake.org/cmake/help/v2.8.12/cmake.html#variable:CMAKE_LANG_VISIBILITY_PRESET

Reply:
In that case I prefer the simpler solution suggested by @lrineau